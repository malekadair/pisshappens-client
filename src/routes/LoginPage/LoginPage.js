import React, { Component } from 'react'
import { Redirect } from 'react-router-dom'
import LoginForm from '../../components/LoginForm/LoginForm'
import TokenService from '../../services/token-service'
import Nav from '../../components/Nav/Nav';
import './LoginPage.css'
import Footer from '../../components/Footer/Footer';


class LoginPage extends Component {
    static defaultProps = {
        location: {},
        history: {
            push: () => { },
        },
    }

    constructor() {
        super()
        this.state = {
            loggedIn: TokenService.hasAuthToken()
                ? true : false
        }
    }

    setLoggedIn = () => {
        this.setState({
            loggedIn: TokenService.hasAuthToken()
                ? true
                : false
        })
    }

    render() {

        return (
            <section className='LoginPage'>
                <header>
                    <Nav />
                </header>
                {this.state.loggedIn ? <Redirect to='/' /> : <LoginForm
                    onLoginSuccess={this.setLoggedIn} />}
                <Footer />
            </section>
        )
    }
}

export default LoginPage