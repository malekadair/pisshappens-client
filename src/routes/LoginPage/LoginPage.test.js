import React from "react";
import toJson from "enzyme-to-json";
import LoginPage from "./LoginPage";

describe(`LoginPage component`, () => {
	const props = {
		id: "a",
		name: "test-class-name"
	};

	it("renders a LoginPage by default", () => {
		const wrapper = <LoginPage />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("renders the LoginPage given props", () => {
		const wrapper = <LoginPage {...props} />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});