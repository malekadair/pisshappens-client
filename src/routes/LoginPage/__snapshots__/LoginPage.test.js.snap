// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renders a LoginPage by default 1`] = `undefined`;

exports[`LoginPage component renders the LoginPage given props 1`] = `undefined`;
